pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -425,8 +425,22 @@ public void addProducer(Producer producer) throws BrokerServiceException {
             }
 
             if (!producers.add(producer)) {
-                throw new NamingException(
-                        "Producer with name '" + producer.getProducerName() + "' is already connected to topic");
+                boolean canOverwrite = false;
+                for (Producer existProducer : producers.values()) {
```

Comment:
Can we avoid iterating over the producers set? I.e. can you change the hash set to a hash map?